### ISSUE TITLE
IsUnion returns false for typeof<Option<int>>

### ISSUE THREAD

**Issue by kerams** [fable-compiler/Fable#2110]
### Description

`Reflection.FSharpType.IsUnion typeof<Option<int>>` returns `false`.

### Expected and actual results
Expecting `true` like on .NET.

### Related information

* Fable version: Online REPL

**Comment by alfonsogarciacaro** [comment]
Yes, in Fable option is a special type and is not like other unions. Although I guess we can fix it in the reflection info, I've to check. Do you have an example of a scenario that's being blocked by this?

**Comment by kerams** [comment]
https://github.com/Zaid-Ajaj/Fable.Remoting/pull/158/files#diff-5bdc4da2122faa430fe9e2f88bdb24e6R425

Not really a blocker since one can always specifically typecheck for Option.

**Comment by MangelMaxime** [comment]
Fixed in https://github.com/fable-compiler/Fable/pull/4529
